core/src/main/scala-3.0.0-RC3/cats/derived/foldable.scala

```diff
@@ -0,0 +1,59 @@
+package cats.derived
+
+import cats.{Eval, Foldable}
+import shapeless3.deriving.{K1, Continue}
+
+object foldable extends FoldableDerivation
+
+trait ProductFoldable[T[x[_]] <: Foldable[x], F[_]] extends Foldable[F]:
+
+  val inst: K1.ProductInstances[T, F]
+
+  def foldLeft[A, B](fa: F[A], b: B)(f: (B, A) => B): B =
+    inst.foldLeft[A, B](fa)(b)(
+      [t[_]] => (acc: B, fd: T[t], t0: t[A]) => Continue(fd.foldLeft(t0, acc)(f))
```

Comment:
Does `Continue` give us stack safety?

Reply:
It does. Imperative loop under the hood.

Reply:
Maybe I'm missing something but I don't see how this can be stack safe due to alternating calls to `ProductInstances.foldLeft` and `CoproductInstances.fold` 🤔

Reply:
Hmm ... yes you're right.

Reply:
Good point! 🤔